src/early_multicellular_stage/Microbe.Multicellular.cs

```diff
@@ -0,0 +1,227 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+using Newtonsoft.Json;
+
+/// <summary>
+///   Early multicellular functionality of the microbes. This is a separate file to put all of the core logic of
+///   the overridden behaviours in the same place.
+/// </summary>
+public partial class Microbe
+{
+    [JsonProperty]
+    private int nextBodyPlanCellToGrowIndex = -1;
+
+    [JsonProperty]
+    private bool enoughResourcesForBudding;
+
+    [JsonProperty]
+    private Dictionary<Compound, float>? compoundsNeededForNextCell;
+
+    [JsonProperty]
+    private Dictionary<Compound, float>? compoundsUsedForMulticellularGrowth;
+
+    [JsonProperty]
+    private Dictionary<Compound, float>? totalNeededForMulticellularGrowth;
+
+    [JsonIgnore]
+    public bool IsFullyGrownMulticellular => nextBodyPlanCellToGrowIndex >= CastedMulticellularSpecies.Cells.Count;
+
+    public void ApplyMulticellularNonFirstCellSpecies(EarlyMulticellularSpecies species, CellType cellType)
+    {
+        cachedMicrobeSpecies = null;
+        cachedMulticellularSpecies = species;
+        MulticellularCellType = cellType;
+
+        Species = species;
+
+        FinishSpeciesSetup();
+    }
+
+    /// <summary>
+    ///   Adds the next cell missing from this multicellular species' body plan to this microbe's colony
+    /// </summary>
+    public void AddMulticellularGrowthCell()
+    {
+        if (Colony == null)
+        {
+            MicrobeColony.CreateColonyForMicrobe(this);
+
+            if (Colony == null)
+                throw new Exception("An issue occured during colony creation!");
+
+            GD.Print("Created a new colony for multicellular cell");
+        }
+
+        var template = CastedMulticellularSpecies.Cells[nextBodyPlanCellToGrowIndex];
+
+        var cell = CreateMulticellularColonyMemberCell(template.CellType);
+
+        // We don't reset our state here in case we want to be in engulf mode
+
+        cell.State = State;
+
+        // Attach the created cell to the right spot in our colony
+        var ourTransform = GlobalTransform;
+
+        var attachVector = ourTransform.origin + Hex.AxialToCartesian(template.Position);
+
+        // TODO: figure out the actually right math for the cell position attach
+        // If we don't adjust like this the cells overlap way too much
+        attachVector *= Constants.MULTICELLULAR_CELL_ATTACH_DISTANCE_MULTIPLIER;
+
+        cell.GlobalTransform = new Transform(
+            MathUtils.CreateRotationForOrganelle(template.Orientation) * ourTransform.basis.Quat(),
```

Comment:
Can we add a comment somewhere around here that OnIGotAddedToMicrobe() method seems to messs up the position? 
Just so we know where to look in case we dont get to fix it this update?

Reply:
actually its GetNewRelativeTransform()

Reply:
I'll definitely put a TODO comment somewhere, and probably makes sense to put it in multiple places, if I don't get this fixed. I'll resume working on this tomorrow.